### PR TITLE
Report deprecation error on functions sha3 and suicide also without call

### LINKED ISSUE #5068: `sha3` on its own does not generate warning
Just `sha3` (without call) does not generate the deprecation warning.

### PR DESCRIPTION
**Pull request by leonardoalt**
Fixes #5068

### CODE REVIEW

File: libsolidity/analysis/TypeChecker.cpp [hunk 1]
```diff
@@ -2639,6 +2619,28 @@ bool TypeChecker::visit(Identifier const& _identifier)
...
 			annotation.isPure = true;
+
+	// Check for deprecated function names.
+	// The check is done here for the case without an actual function call.
+	if (
+		_identifier.annotation().type &&
+		_identifier.annotation().type->category() == Type::Category::Function
+	)
+	{
+		FunctionType const* fType = dynamic_cast<FunctionType const*>(_identifier.annotation().type.get());
```
Comment (chriseth) [inline libsolidity/analysis/TypeChecker.cpp]: Please use dynamic_cast with reference type.

**Comment by leonardoalt** [comment]
Updated

File: libsolidity/analysis/TypeChecker.cpp [hunk 2]
```diff
@@ -2639,6 +2619,27 @@ bool TypeChecker::visit(Identifier const& _identifier)
 	else if (dynamic_cast<MagicVariableDeclaration const*>(annotation.referencedDeclaration))
 		if (dynamic_cast<FunctionType const*>(annotation.type.get()))
 			annotation.isPure = true;
+
+	// Check for deprecated function names.
+	// The check is done here for the case without an actual function call.
+	if (
```
Comment (chriseth) [inline libsolidity/analysis/TypeChecker.cpp]: Actually this whole thing can be shorted to
```suggestion
	if (FunctionType const* fType = dynamic_cast<FunctionType const*>(_identifier.annotation().type.get()))
```

**Review by chriseth — DISMISSED** [review]
Can be merged with or without the latest suggested change.

**Review by chriseth — APPROVED** [review]

**Pull request merged**